nav2_map_server/README.md

```diff
@@ -1,59 +1,137 @@
+template<>
+class MapServer<sensor_msgs::msg::PointCloud2> : public nav2_util::LifecycleNode {...};
+
+// map_3d/map_saver_3d.hpp
+// map_3d/map_saver_3d.cpp
+template<>
+class MapSaver<sensor_msgs::msg::PointCloud2> : public nav2_util::LifecycleNode {...};
 ```
 
-Where the YAML file specified contained the various map metadata, such as:
+### Library interfaces list
+
+The following list provides the headers mapped to their present functionalities. *For example if one wants to use  the server node then adding `#include "map_server/map_server.hpp"` will provide the server node implementation with all the specializations for different map type.*
 
+| Basic Components | Functionality | Namespace | Contains
+| :-----------------: | :-----------------: | :------------------: | :------------: |
+| map_server.hpp | Template Map Server | nav2_map_server | `MapServer<OccupancyGrid>()`<br /> `MapServer<Pointcloud2>()`
+| map_saver.hpp | Template Map Saver | nav2_map_server | `MapSaver<OccupancyGrid>()`<br /> `MapSaver<Pointcloud2>()`
+| map_io_2d.hpp | 2D Map IO | nav2_map_server::map_2d | Functions for OccupancyGrid I/O operations
+| map_io_3d.hpp | 3D Map IO | nav2_map_server::map_3d | Functions for PointCloud2 I/O operations
+
+### Existing Executables
+
+The existing system of Nav2 Map Server provides many executables to get a hassle free services launch and handling. Listed below all the executables with their callable names, and functionalities:
+
+| Executables | Implementation | Functionality | Type | Map Type
+| :-------------: | :-----------------: | :--------------: | :----: | :--- |
+| map_server_2d | main_2d.cpp | Map Server | LifecycleNode | OccupancyGrid
+| map_server_3d | main_3d.cpp | Map Server | LifecycleNode | PointCloud2
+| map_saver_server_2d | main_server_2d.cpp | Map Saver Server | LifecycleNode | OccupancyGrid
+| map_saver_server_3d | main_server_3d.cpp | Map Saver Server | LifecycleNode | PointCloud2
+| map_saver_cli | main_cli.cpp | Map Saver CLI | CLI | Detects based<br /> on input parameters
+
+These executables are used in running the `server/saver_server` with ros2 launch system, or sometimes used as a direct CLI, e.g `map_saver_cli` is used as a `CLI` to save map from a topic to file, on contrary to this `map_saver_server_*` is used with launch system and the launch handles the run of a `LifecycleManager` in order to get the `saver_server` configured, same for the `map_server_*`. 
+
+**Note:** * Although there is no launch file directly provided by `Nav2 Map Server` for launching a `map_server_*`, instead the external processes used to run the `map_server_*` according to their usage requirements, there can be multiplle instances of `map_server_*`, and that should be usage specific. [For example  view:  nav2_bringup/bringup/launch/localization_launch.py](https://github.com/ShivamPR21/navigation2/blob/de73c104f1d2407984c6de0e03c6ccf4d1d09f65/nav2_bringup/bringup/launch/localization_launch.py#L85)*
```

Comment:
Please do not refer to the specific lines to any code place. This will be difficult to maintain and give us a lot of boilerplate work when this file will be changed.
I think this might be simplified to something like:
->
**Note:** Although there is no launch file directly executing `map_server`, its launching principles are the same as for `map_saver_server` and could be found in a `nav2_bringup/bringup/launch/localization_launch.py` file.

Also, this note is better be placed at "Service Usage" section right after `$ ros2 launch nav2_map_server map_saver_server_3d.launch.py` line to make a continuous and seamless story.